documentation/src/main/antora/modules/ROOT/pages/component-checkpoint.adoc

```diff
@@ -0,0 +1,279 @@
+
+While reading records (`next()` method), the connector checks whether a checkpoint can be provided.
+
+If necessary, the checkpoint is generated by calling `getCheckpoint()`, which internally calls the method annotated with `@CheckpointData`.
+
+[source,java]
+----
+  final Consumer<CheckpointState> checkpointCallback = bookmark -> {
+    serializeCheckpoint(bookmark);
+  };
+  ...
+  input.start(checkpointCallback);
+  Record record;
+  while ((record = input.next()) != null) {
+    // process records
+  }
+  input.stop();
+----
+
+
+== How to resume
+
+As state previously, checkpoints are configured via the `@Checkpoint` annotated configuration class. The `@Checkpoint` annotation can be used to specify the method type used for checkpointing and the checkpointing frequency.
+The checkpointing frequency can be set to `RECORD` or `TIME`. The `RECORD` frequency saves the checkpoint after processing a certain number of records, while the `TIME` frequency saves the checkpoint after a certain time interval.
+
+When the component is restarted, the runtime calls the `start()` method. The connector then retrieves the checkpoint object and resumes processing from the last saved state from the configuration.
+
+When implementing checkpointing, it is recommended to use a separate configuration class for checkpointing. This class should be nested within the main configuration class and annotated with `@Checkpoint`. This ensures that the checkpointing configuration is separate from the main configuration and can be easily managed by the runtime.
+
+
+Sample configuration class with checkpointing annotations:
+[source,java]
+----
+    ...
+    @Checkpoint
+    @Version(value = 2, migrationHandler = CheckpointMigrationHandler.class)
+    public static class CheckPointInputConfig implements Serializable {
+
+        public enum Strategy {
+            BY_ID,
+            BY_DATE
+        }
+
+        @Option
+        @DefaultValue("BY_ID")
+        private Strategy strategy = Strategy.BY_ID;
+
+        @Option
+        private String startDate;
+
+        @Option
+        private int sinceId;
+
+        @Option
+        private String status;
+    }
+
+    @Data
+    @GridLayout(value = { @GridLayout.Row("user"), @GridLayout.Row("pass")})
+    @GridLayout(names = GridLayout.FormType.CHECKPOINT, value = { @GridLayout.Row("checkPointInputConfig") })
+    public static class InputConfig {
+
+        @Option
+        private String user;
+
+        @Option
+        private String pass;
+
+        @Option
+        private CheckPointInputConfig checkpoint = new CheckPointInputConfig();
+    }
+----
+
+Calling the `getCheckpoint()` returns `CheckpointState` class, which is a simple POJO class that holds the checkpoint data and version. This class has to be serializable to allow the runtime to save and restore the checkpoint object. For that, it provides a helper method `toJson()` to serialize the checkpoint state object.
```

Comment:
Can't see any CheckpointState  class in the example.

Reply:
In section 2 of the document.